src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -1013,6 +1013,12 @@ impl Step for Rustc {
     }
 
     fn make_run(run: RunConfig<'_>) {
+        // If only `compiler` was passed, do not run this step.
+        // Instead the `Assemble` step will take care of compiling Rustc.
+        if run.builder.paths == vec![PathBuf::from("compiler")] {
+            return;
+        }
```

Comment:
Remark: this kinda feels iffy, in that I feel like the path filtering/handling logic that lead to

>  `Rustc` step was invoked twice, once with all crates and once with no crates (even though both of these mean the same thing).

makes this kinda weird.

Reply:
Yeah, that does make sense.